Config with default template

gforcada opened the pull request.

mister-roboto commented:
@gforcada thanks for creating this Pull Request and helping to improve Plone!

TL;DR: Finish pushing changes, pass all other checks, then paste a comment:
```
@jenkins-plone-org please run jobs
```

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass, but it takes 30-60 min.  Other CI checks are usually much faster and the Plone Jenkins resources are limited, so when done pushing changes and all other checks pass either [start all Jenkins PR jobs yourself](https://jenkinsploneorg.readthedocs.io/en/latest/run-pull-request-jobs.html#run-a-pull-request-job), or simply add the comment above in this PR to start all the jobs automatically.

Happy hacking!

gforcada commented:
According to `sources.cfg` in `buildout.coredev` for branch `5.2` we use `plone.rest` branch `1.x`.

This PR is meant for master branch, yet, it is still being tested against Plone `5.2`. Is that intended? Where's the mistake, on `buildout.coredev` or the GHA matrix defined here? 🤔

davisagli commented:
@gforcada I don't think there is a mistake. plone.rest 1.x is the release series that is offically part of Plone 5.2, because it was the latest major release when Plone 5.2 was released. But more recent major versions of plone.rest still support being used with Plone 5.2 as an opt-in upgrade to get the /++api++ traverser.

gforcada commented:
Thanks @davisagli then I removed all plone 5.2 and python 2.7 references 👍🏾

davisagli commented:
@gforcada No, sorry, I don't think I got my message across. I was trying to say that it is intentional that the master branch of plone.rest continues to support Plone 5.2, even though it is not the version that is pinned by default on Plone 5.2.

Personally I think it would be okay to remove support for Plone 5.2 if we bump the major version number of plone.rest. Not sure if @tisto or someone else feels differently...

gforcada commented:
Oh, sorry 😅 indeed, I read it totally wrong 🤦🏾 

Let's see what the others think.

tisto commented:
@gforcada plone.rest 3.x still supports Python 2.7 and Plone 5.2. plone.rest and plone.restapi are meant to be the "abstraction layer" around Plone, which abstracts away differences between different Plone versions. This allows Volto and other consumer applications to move at a different pace and avoids tight coupling between the frontend and the backend. 

During the development of Volto, this was a necessity. Otherwise, Plone 6 with Volto would never have happened in the way it did. We would still be stuck with Plone 5.2 + Volto.

If we want Volto to continue to be able to move at a faster pace and the backend to stay stable, we have to keep it that way IMHO. Plone 6.1 and Plone 7 will happen and we don't want to end up with lots of different branches on the backend that barely differ. This would mean we would have to backport features and bugfixes across multiple branches leading to lots of unnecessary work.

This is why we decided that those two packages have their own release cycle and release manager and abstract away the differences. This also means that the repositories need a slightly different structure that supports testing multiple Plone versions at the same time.

I am +1 to moving those two packages closer to the best practices of the other core packages. Though, we need to consider the whole picture and maybe build a consensus in the community on which way we want to go in the future.

It is a bit unfortunate that I just cut a 3.0.0 release where we cleaned up most things and did not drop Python 2.7. My thinking behind this was that plone.rest is even a more "lower-level" package, so we should be even more conservative. In plone.restapi 8.0.0, we dropped Python 2.7 support but kept Plone 5.2.

This is one of the topics that we scheduled for Beethoven Sprint, so this might be a good opportunity to discuss this in person (https://kitconcept.com/plone-beethoven-sprint-bonn-2023/). Maybe you want to consider coming. :)

tisto commented:
@gforcada FYI: the current state of Python/Plone support is documented in the README:

https://github.com/plone/plone.rest#plonepython-support

gforcada commented:
@tisto yes, I'm aware of `plone.rest` and `restapi` being _special_, if keeping python 2.7 or not makes sense is another matter (in 2023, 3 years after 2.7 sunset and with restapi already having dropped 2.7 support, you probably have much bigger problems than if the _newest_ version of `plone.rest` is still compatible with python 2.7 🤷🏾 

Discussing it in the sprint, yes, that would be great, seems that it is fully booked though 😕 me and my two other team mates would really want to attend, but never got any news/tip about where to register, although I asked @sneridagh since Alpine sprint 😅 

Building consensus with best practices, I'm all for it, so far, the best practices, and specially _how to apply_ those best practices are in https://github.com/plone/meta/ if you have better alternatives, issues and pull requests are welcome. There are plenty of [further ideas](https://github.com/plone/meta/issues) to expand from the very minimal set that we are applying right now across all repositories.

Trying to move this PR forward, deleting the [last commit](https://github.com/plone/plone.rest/pull/150/commits/bea257e8bb27cfd8cfb947a0ab20f7ffcef71cce) would be enough to get this PR approved, or are there other changes that would be required to get this merged? 🤔 

A review would be nice ✨

tisto commented:
@gforcada I'd be ok with dropping Python 2 compatibility if keeping it is too much work. As said, it was just a bit unfortunate that I just cut a major release without dropping Python 2 compatibility. I will try to find time to check the meta package and the new package layout and then get back to you with a review...

tisto commented:
@gforcada I dropped Python 2 support with plone.rest 4.0.0. We can re-visit making plone.rest compliant with the Plone core best practices. Maybe something we can discuss in two weeks at ploneconf?

gforcada commented:
I rebased the fixed the PR, let's see what CI / Jenkins think of it 🍀

gforcada commented:
I removed two commits that were probably the source of why the 5.2 job was breaking.

And given that `plone.rest` already has a quite elaborate CI setup, I removed the test and coverage CI jobs from `plone/meta`, so we have best of both worlds:

- all the QA checks and common DX from `plone/meta`
- the tailored testing setup from `plone.rest`

Let's see if GHA and Jenkins like the changes 🍀

gforcada commented:
@jenkins-plone-org please run jobs

tisto commented on news/93e1ab65.internal:
```diff
@@ -0,0 +1,2 @@
+Update configuration files.
```
@gforcada is the name of the file autogenerated or where does 93e1ab65 come from?

gforcada replied:
Yes, autogenerated, a commit hash, or directly something random, no idea, but somewhere in `plone/meta` has the logic for that.

We can modify that to something else if you want

tisto commented on setup.py:
```diff
@@ -53,22 +53,26 @@ def read(*rnames):
     install_requires=[
         "setuptools",
         "collective.monkeypatcher",
-        "zope.component",
```
@gforcada do I understand correctly that we are not explicit about the libraries that we use in install_requires any longer? Any dependency of Products.CMFCore will not be listed here in the future, right? I don't have a strong opinion here. I'd just like to understand our current best practices, so I can stick to them.

wesleybl replied:
@gforcada interesting. I didn't know about this problem with the delay in installation. So this is yet another reason to use the Plone package dependencies.

My suggestion for improvement is also what I said in the previous comment, to try to do this automatically. Dependencies always change.

We also have the problem of packages that are compatible with Plone 6.0 and 6.1. If a package is no longer a dependency on Plone 6.1, but remains a dependency on Plone 6.0, it should be in the install_requires of a package that is compatible with Plone 6.0 and 6.1. So the list of ignored packages should contain all dependencies of both versions of Plone. And in the future include Plones 6.2, 7.0... This is difficult to maintain. So if there is a way for this to be automatic it would be wonderful.

gforcada replied:
`pip` being a bit slower to install, is not so much important, specially in the day of getting things CI/CD that you don't have to wait on your local computer to do something else afterwards. It is still noticeable, but not too big of a deal ⚖️ 

Getting `z3c.dependencychecker` (or any other dependency checker for that matter) smarter would be great indeed, ideas/PRs welcome 😄 https://github.com/reinout/z3c.dependencychecker/

Handle 6.0/1 dependencies, and even 7.0 at some point... that's hard... something that could help a bit would be https://github.com/reinout/z3c.dependencychecker/issues/106 i.e. try to be smart when detecting a new dependency and see if it is wrapped in a `try/except` block and thus do not consider it a hard dependency...

But that's hairy to get a specific answer for all cases.

tisto replied:
@gforcada thanks for elaborating on this. I think it makes sense to automate those kind of things as much as possible. Automation and consistency beat 100% accuracy.

tisto reviewed (approved):
Apart from my questions in the PR, this LGTM.

gforcada commented:
@tisto it's _your_ package, so I let you merge it if the explanations above are good enough for you

tisto commented:
Thank you @gforcada for your help on this!
